src/FilenameParsing/HashFileIDHelper.php

```diff
@@ -14,70 +13,13 @@
  *
  * e.g.: `Uploads/a1312bc34d/sam__ResizedImageWzYwLDgwXQ.jpg`
  */
-class HashFileIDHelper implements FileIDHelper
+class HashFileIDHelper extends AbstractFileIDHelper
 {
-    use Injectable;
-
     /**
      * Default length at which hashes are truncated.
      */
     const HASH_TRUNCATE_LENGTH = 10;
 
-    public function buildFileID($filename, $hash = null, $variant = null, $cleanfilename = true)
-    {
-        if ($filename instanceof ParsedFileID) {
-            $hash =  $filename->getHash();
-            $variant =  $filename->getVariant();
-            $filename =  $filename->getFilename();
-        }
-
-        if (empty($hash)) {
-            throw new InvalidArgumentException('HashFileIDHelper::buildFileID requires an $hash value.');
-        }
-
-        // Since we use double underscore to delimit variants, eradicate them from filename
-        if ($cleanfilename) {
-            $filename = $this->cleanFilename($filename);
-        }
-        $name = basename($filename ?? '');
-
-        // Split extension
-        $extension = null;
-        if (($pos = strpos($name ?? '', '.')) !== false) {
-            $extension = substr($name ?? '', $pos ?? 0);
-            $name = substr($name ?? '', 0, $pos);
-        }
-
-        $fileID = $this->truncate($hash) . '/' . $name;
-
-        // Add directory
-        $dirname = ltrim(dirname($filename ?? ''), '.');
-        if ($dirname) {
-            $fileID = $dirname . '/' . $fileID;
-        }
-
-        // Add variant
-        if ($variant) {
-            $fileID .= '__' . $variant;
-        }
-
-        // Add extension
-        if ($extension) {
-            $fileID .= $extension;
-        }
-
-        return $fileID;
-    }
-
-    public function cleanFilename($filename)
```

Comment:
Moved this method to the abstract class as it's identical to the implementation in `NaturalFileIDHelper`